packages/superset-ui-encodable/src/types/Encoding.ts

```diff
@@ -0,0 +1,27 @@
+import { ChannelType, ChannelTypeToDefMap } from './Channel';
+import { Value } from './VegaLite';
+import ChannelEncoder from '../encoders/ChannelEncoder';
+
+export type EncodingConfig = {
+  [k in string]: [ChannelType, Value, 'multiple'?];
+};
+
+export type DeriveChannelTypes<Config extends EncodingConfig> = {
+  readonly [k in keyof Config]: Config[k]['0'];
```

Comment:
`#TIL`, did not know that you could index types like this. 

interesting it has to be a string and not a number.